app/services/account_builder.rb

```diff
@@ -164,13 +164,11 @@ def build_account_users
   # param is present, and the affiliate exists. Also ensure affiliate_other
   # gets wiped out if the affiliate_id does not point to `Other`.
   def set_affiliate
-    if account.class.using_affiliate? && account_params.key?(:affiliate_id)
-      affiliate = Affiliate.find_by_id(account_params[:affiliate_id])
-      account.affiliate_id = affiliate.try(:id)
-      account.affiliate_other = nil if affiliate.try(:id) != Affiliate.OTHER.id
+    if affiliate.present?
+      account.affiliate_id = affiliate.id
+      account.affiliate_other = affiliate.subaffiliates_enabled? ? account_params[:affiliate_other] : nil
```

Comment:
Here's a thought. Could we move this check into `Affiliate` itself?

```
def affiliate_other=(value)
  self[:affiliate_other] = value if subaffiliates_enabled?
end
```

Or it could assign `nil` if enabled is false.